lib/puppet/module_tool/metadata.rb

```diff
@@ -203,19 +203,19 @@ def validate_data_provider(value)
       err = nil
       if value.is_a?(String)
         unless value =~ /^[a-zA-Z][a-zA-Z0-9_]*$/
-          err = value =~ /^[a-zA-Z]/ ? 'contains non-alphanumeric characters' : 'must begin with a letter'
+          err = value =~ /^[a-zA-Z]/ ? _('contains non-alphanumeric characters') : _('must begin with a letter')
         end
       else
-        err = 'must be a string'
+        err = _('must be a string')
       end
-      raise ArgumentError, "field 'data_provider' #{err}" if err
+      raise ArgumentError, _("field 'data_provider' %{err}") % { err: err } if err
```

Comment:
could you add a note that `'data_provider'` is a metadata option and that this line is the beginning of the concatenation of one of the two preceding lines?